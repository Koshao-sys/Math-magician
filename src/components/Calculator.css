.key-area {
  display: grid;
  grid-template-columns: auto auto auto auto;
  grid-template-rows: auto auto auto auto auto;
}