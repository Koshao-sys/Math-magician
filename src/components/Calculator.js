import React from 'react';
import { v4 as uuidv4 } from 'uuid';
import './Calculator.css';

function Display() {
  return (
    <div className="DisplayArea">
      <p>0</p>
    </div>
  );
}

function KeyArea() {
  const Items = ['AC', '+/-', '%', 'รท', '7', '8', '9', 'x', '4', '5', '6', '-', '1', '2', '3', '+', '0', '.', '='];
  return (
    <div className="key-area">
      {Items.map((Item, ind) => (
        <button key={uuidv4()} id={`btn-${ind}`} type="submit" className="key-btn">{Item}</button>
      ))}
    </div>
  );
}

function Calculator() {
  return (
    <div className="Calculator">
      <Display />
      <KeyArea />
    </div>
  );
}

export default Calculator;
