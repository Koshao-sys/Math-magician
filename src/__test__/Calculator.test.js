import { render } from '@testing-library/react';
import Calculator from '../components/Calculator';

it('renders correctly', () => {
    const calculator = render(<Calculator />);
    expect(calculator).toMatchSnapshot();
});