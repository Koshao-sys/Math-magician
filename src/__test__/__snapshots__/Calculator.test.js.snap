// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="calc-container"
      >
        <div
          class="Calculator"
        >
          <div
            class="DisplayArea"
          >
            <p>
              0
            </p>
          </div>
          <div
            class="key-area"
          >
            <button
              class="key-btn"
              id="btn-0"
              type="submit"
            >
              AC
            </button>
            <button
              class="key-btn"
              id="btn-1"
              type="submit"
            >
              +/-
            </button>
            <button
              class="key-btn"
              id="btn-2"
              type="submit"
            >
              %
            </button>
            <button
              class="key-btn"
              id="btn-3"
              type="submit"
            >
              รท
            </button>
            <button
              class="key-btn"
              id="btn-4"
              type="submit"
            >
              7
            </button>
            <button
              class="key-btn"
              id="btn-5"
              type="submit"
            >
              8
            </button>
            <button
              class="key-btn"
              id="btn-6"
              type="submit"
            >
              9
            </button>
            <button
              class="key-btn"
              id="btn-7"
              type="submit"
            >
              x
            </button>
            <button
              class="key-btn"
              id="btn-8"
              type="submit"
            >
              4
            </button>
            <button
              class="key-btn"
              id="btn-9"
              type="submit"
            >
              5
            </button>
            <button
              class="key-btn"
              id="btn-10"
              type="submit"
            >
              6
            </button>
            <button
              class="key-btn"
              id="btn-11"
              type="submit"
            >
              -
            </button>
            <button
              class="key-btn"
              id="btn-12"
              type="submit"
            >
              1
            </button>
            <button
              class="key-btn"
              id="btn-13"
              type="submit"
            >
              2
            </button>
            <button
              class="key-btn"
              id="btn-14"
              type="submit"
            >
              3
            </button>
            <button
              class="key-btn"
              id="btn-15"
              type="submit"
            >
              +
            </button>
            <button
              class="key-btn"
              id="btn-16"
              type="submit"
            >
              0
            </button>
            <button
              class="key-btn"
              id="btn-17"
              type="submit"
            >
              .
            </button>
            <button
              class="key-btn"
              id="btn-18"
              type="submit"
            >
              =
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="calc-container"
    >
      <div
        class="Calculator"
      >
        <div
          class="DisplayArea"
        >
          <p>
            0
          </p>
        </div>
        <div
          class="key-area"
        >
          <button
            class="key-btn"
            id="btn-0"
            type="submit"
          >
            AC
          </button>
          <button
            class="key-btn"
            id="btn-1"
            type="submit"
          >
            +/-
          </button>
          <button
            class="key-btn"
            id="btn-2"
            type="submit"
          >
            %
          </button>
          <button
            class="key-btn"
            id="btn-3"
            type="submit"
          >
            รท
          </button>
          <button
            class="key-btn"
            id="btn-4"
            type="submit"
          >
            7
          </button>
          <button
            class="key-btn"
            id="btn-5"
            type="submit"
          >
            8
          </button>
          <button
            class="key-btn"
            id="btn-6"
            type="submit"
          >
            9
          </button>
          <button
            class="key-btn"
            id="btn-7"
            type="submit"
          >
            x
          </button>
          <button
            class="key-btn"
            id="btn-8"
            type="submit"
          >
            4
          </button>
          <button
            class="key-btn"
            id="btn-9"
            type="submit"
          >
            5
          </button>
          <button
            class="key-btn"
            id="btn-10"
            type="submit"
          >
            6
          </button>
          <button
            class="key-btn"
            id="btn-11"
            type="submit"
          >
            -
          </button>
          <button
            class="key-btn"
            id="btn-12"
            type="submit"
          >
            1
          </button>
          <button
            class="key-btn"
            id="btn-13"
            type="submit"
          >
            2
          </button>
          <button
            class="key-btn"
            id="btn-14"
            type="submit"
          >
            3
          </button>
          <button
            class="key-btn"
            id="btn-15"
            type="submit"
          >
            +
          </button>
          <button
            class="key-btn"
            id="btn-16"
            type="submit"
          >
            0
          </button>
          <button
            class="key-btn"
            id="btn-17"
            type="submit"
          >
            .
          </button>
          <button
            class="key-btn"
            id="btn-18"
            type="submit"
          >
            =
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
